src/msg_handler.c

```diff
@@ -0,0 +1,388 @@
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <ctype.h>
+
+#include "msg_handler.h"
+#include "global_vars.h"
+#include "output.h"
+#include "led.h"
+#include "fan.h"
+#include "buzzer.h"
+
+static char* itoa(int value, char* string);
+
+int parse_msg(char* buf, char* response, int max_len)
+{
+    char temp_str[11] = {0};  // for doing itoa conversions
+    response[0] = '\0';  // make a blank string
```

Comment:
```suggestion
    if (max_len == 0) return;
    response[0] = '\0';  // make a blank string
```
I know we wouldn't expect `max_len` to ever be 0 in practice, but this is a cheap safety check and helps stop a bad problem getting any worse.

EDIT: Pls disregard, I now see that the function's contract assumes `response` can accept at least `max_len+1` bytes, so this is safe.